samples/extensions/graphics_pipeline_library/graphics_pipeline_library.cpp

```diff
@@ -0,0 +1,539 @@
+	shader_module_create_info.pCode    = spirv.data();
+
+	VkPipelineShaderStageCreateInfo shader_Stage_create_info{};
+	shader_Stage_create_info.sType = VK_STRUCTURE_TYPE_PIPELINE_SHADER_STAGE_CREATE_INFO;
+	shader_Stage_create_info.pNext = &shader_module_create_info;
+	shader_Stage_create_info.stage = VK_SHADER_STAGE_FRAGMENT_BIT;
+	shader_Stage_create_info.pName = "main";
+
+	// Select lighting model using a specialization constant
+	srand((unsigned int) time(NULL));
+	uint32_t lighting_model = (int) (rand() % 3);
+
+	// Each shader constant of a shader stage corresponds to one map entry
+	VkSpecializationMapEntry specialization_map_entry{};
+	specialization_map_entry.constantID = 0;
+	specialization_map_entry.size       = sizeof(uint32_t);
+
+	VkSpecializationInfo specialization_info{};
+	specialization_info.mapEntryCount = 1;
+	specialization_info.pMapEntries   = &specialization_map_entry;
+	specialization_info.dataSize      = sizeof(uint32_t);
+	specialization_info.pData         = &lighting_model;
+
+	shader_Stage_create_info.pSpecializationInfo = &specialization_info;
+
+	VkGraphicsPipelineCreateInfo pipeline_library_create_info{};
+	pipeline_library_create_info.sType              = VK_STRUCTURE_TYPE_GRAPHICS_PIPELINE_CREATE_INFO;
+	pipeline_library_create_info.pNext              = &library_info;
+	pipeline_library_create_info.flags              = VK_PIPELINE_CREATE_LIBRARY_BIT_KHR | VK_PIPELINE_CREATE_RETAIN_LINK_TIME_OPTIMIZATION_INFO_BIT_EXT;
+	pipeline_library_create_info.stageCount         = 1;
+	pipeline_library_create_info.pStages            = &shader_Stage_create_info;
+	pipeline_library_create_info.layout             = pipeline_layout;
+	pipeline_library_create_info.renderPass         = render_pass;
+	pipeline_library_create_info.pDepthStencilState = &depth_stencil_state;
+	pipeline_library_create_info.pMultisampleState  = &multisample_state;
+
+	VkPipeline fragment_shader = VK_NULL_HANDLE;
+	VK_CHECK(vkCreateGraphicsPipelines(get_device().get_handle(), thread_pipeline_cache, 1, &pipeline_library_create_info, nullptr, &fragment_shader));
+
+	// Create the pipeline using the pre-built pipeline library parts
+	// Except for above fragment shader part all parts have been pre-built and will be re-used
+	std::vector<VkPipeline> libraries = {
+	    pipeline_library.vertex_input_interface,
+	    pipeline_library.pre_rasterization_shaders,
+	    fragment_shader,
+	    pipeline_library.fragment_output_interface};
+
+	// Link the library parts into a graphics pipeline
+	VkPipelineLibraryCreateInfoKHR linking_info{};
+	linking_info.sType        = VK_STRUCTURE_TYPE_PIPELINE_LIBRARY_CREATE_INFO_KHR;
+	linking_info.libraryCount = static_cast<uint32_t>(libraries.size());
+	linking_info.pLibraries   = libraries.data();
+
+	VkGraphicsPipelineCreateInfo executable_pipeline_create_info{};
+	executable_pipeline_create_info.sType = VK_STRUCTURE_TYPE_GRAPHICS_PIPELINE_CREATE_INFO;
+	executable_pipeline_create_info.pNext = &linking_info;
```

Comment:
executable_pipeline_create_info.layout must not be left NULL see https://www.khronos.org/registry/vulkan/specs/1.3-extensions/html/vkspec.html#VUID-VkGraphicsPipelineCreateInfo-flags-06729 and https://www.khronos.org/registry/vulkan/specs/1.3-extensions/html/vkspec.html#VUID-VkGraphicsPipelineCreateInfo-flags-06730. This came about due a recent spec change: https://gitlab.khronos.org/vulkan/vulkan/-/merge_requests/5164